Tasks/Src/RTOS_BPS.c

```diff
@@ -58,9 +58,49 @@ void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
     OSMutexPost(mutex, options, &err);
     assertOSError(err);
 }
-void RTOS_BPS_TaskCreate(void){
 
+/**
+ * @brief   Creates a task that will be handled by the OS
+ * @param   *p_tcb - pointer to the tcb
+ * @param   *p_name - pointer to task name
+ * @param   p_task - pointer to the task
+ * @param   *p_args - pointer to task function arguments
+ * @param   *prio - task priority
+ * @param   *p_stk_base - the stack
+ * @param   stk_size - size of the stack
+ * @param   *p_err - return error code
+ * @return  nothing to see here
+ */
+void RTOS_BPS_TaskCreate(
+    OS_TCB              *p_tcb,
+    BPS_OS_TASK_NAME    *p_name,
+    BPS_OS_TASK_PTR      p_task,
+    void                *p_arg,
+    BPS_OS_TASK_PRIO     prio,
+    CPU_STK             *p_stk_base,
+    BPS_OS_TASK_STK_SIZE stk_size
+    )
+    {
+        BPS_OS_ERR err;
+        OS_TCB *tcbptr = calloc(1, sizeof(OS_TCB));
```

Comment:
Do not use any memory allocation functions. Micrium does not guarantee these will work at all, and usually doing dynamic memory allocations on embedded systems is a nightmare and makes things explode. There's got to be a different way to do this.

Reply:
https://docs.silabs.com/micrium/latest/micrium-general-concepts/04-memory-segments-and-lib-heap

Instead of calloc/malloc would using the memory allocation functions provided by the OS work? 
This way I can still reduce the amount of parameters.

Reply:
Nah, tis too much work. I'll just leave as is without the calloc calls

Reply:
You should just be using your function parameters, not allocating any memory yourself. Remove the *tcbptr = calloc and the *stkptr = calloc, and just use the tcbptr and stkptr you pass into your function.
`OS_TCB              *p_tcb,
    BPS_OS_TASK_NAME    *p_name,
    BPS_OS_TASK_PTR      p_task,
    void                *p_arg,
    BPS_OS_TASK_PRIO     prio,
    CPU_STK             *p_stk_base,
    BPS_OS_TASK_STK_SIZE stk_size`
    these ones